File: 30-Days-of-Code/Day-7-Arrays/Erlang/solution.erl
-module(solution).
-import(lists,[reverse/1]). 
-export([main/0]).



main() ->
    {N, _} = string:to_integer(string:chomp(io:get_line(""))),

    ArrTemp = re:split(string:chomp(io:get_line("")), "\\s+", [{return, list}, trim]),

    Arr = lists:map(fun(X) -> {I, _} = string:to_integer(X), I end, ArrTemp),

    format_list(reverse(Arr)),

    ok.


fnl([H]) ->
        io:format("~p", [H]);
    fnl([H|T]) ->
        io:format("~p ", [H]),
        fnl(T);
    fnl([]) ->
        ok.

format_list(L) when list(L) ->
        fnl(L).

File: 30-Days-of-Code/Day-7-Arrays/Haskell/solution.hs
{-# LANGUAGE FlexibleInstances, UndecidableInstances, DuplicateRecordFields #-}

module Main where

import Control.Monad
import Data.Array
import Data.Bits
import Data.List
import Data.List.Split
import Data.Set
import Debug.Trace
import System.Environment
import System.IO
import System.IO.Unsafe



main :: IO()
main = do
    n <- readLn :: IO Int

    arrTemp <- getLine

    let arr = Data.List.map (read :: String -> Int) . words $ arrTemp

    let list_to_string = unwords . Data.List.map show

    putStrLn $ list_to_string $ reverse(arr)
